src/test.js

```diff
@@ -40,7 +40,8 @@ var resetTimeout = function () {
     clearTimeout(self.timer);
   }
   self.timer = setTimeout(function () {
-    done.apply(self, [new Error("Timed out")]);
+    const errorMsg = self.async ? "Timed out. Argument length mismatch between step text and function, Ensure done callback is called to mark completion of step." : "Timed out";
```

Comment:
```suggestion
    const errorMsg = self.async ? "Number of parameters in the step do not match the number of arguments in the step definition" : "Timed out";
```

Reply:
Is there a way to throw the parameter mismatch without timing out?

Reply:
We thought about that, doing it at validation phase requires lot more changes as changes has to be done at gauge and gauge-proto too. Other languages like dotnet and java also does this at execution time. As JS is dynamically typed it is not possible to find until it times out during execution.

Reply:
I feel the message has to say that the step has timed out, since it can timeout even when the done callback is called in the step and it actually took more time.

Reply:
Ok. can you change it to

`Timed out. Number of parameters in the step do not match the number of arguments in the step definition`